utilities/backupable/backupable_db.cc

```diff
@@ -1733,25 +1733,48 @@ Status BackupEngineImpl::BackupMeta::StoreToFile(bool sync) {
   if (!app_metadata_.empty()) {
     std::string hex_encoded_metadata =
         Slice(app_metadata_).ToString(/* hex */ true);
+
+	// +1 to accomodate newline character
+	size_t hex_meta_strlen = kMetaDataPrefix.ToString().length() + hex_encoded_metadata.length() + 1;
+	if (hex_meta_strlen >= buf_size) {
+		return Status::Corruption("Buffer too small to fit backup metadata");
+	}
+	else if (len + hex_meta_strlen >= buf_size) {
+		backup_meta_file->Append(Slice(buf.get(), len));
+		buf.release();
+		unique_ptr<char[]> new_reset_buf(new char[max_backup_meta_file_size_]);
+		buf.swap(new_reset_buf);
+		len = 0;
+	}
     len += snprintf(buf.get() + len, buf_size - len, "%s%s\n",
                     kMetaDataPrefix.ToString().c_str(),
                     hex_encoded_metadata.c_str());
-    if (len >= buf_size) {
-      return Status::Corruption("Buffer too small to fit backup metadata");
-    }
+  }
+
+  char writelen_temp[19];
+  if (len + sprintf(writelen_temp, "%" ROCKSDB_PRIszt "\n", files_.size()) >= buf_size) {
+	  backup_meta_file->Append(Slice(buf.get(), len));
+	  buf.release();
+	  unique_ptr<char[]> new_reset_buf(new char[max_backup_meta_file_size_]);
+	  buf.swap(new_reset_buf);
+	  len = 0;
   }
   len += snprintf(buf.get() + len, buf_size - len, "%" ROCKSDB_PRIszt "\n",
                   files_.size());
-  if (len >= buf_size) {
-    return Status::Corruption("Buffer too small to fit backup metadata");
-  }
+
   for (const auto& file : files_) {
     // use crc32 for now, switch to something else if needed
+
+	size_t newlen = len + file->filename.length() + sprintf(writelen_temp, " crc32 %u\n", file->checksum_value);
```

Comment:
can we store the `"%s crc32 %u\n"` in a const variable since it is repeated below? Same for the `"%" ROCKSDB_PRIszt "\n"` above